Mathlib/Data/Sigma/Lex.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Data.Sigma.Basic
+import Mathlib.Order.RelClasses
+
+/-!
+# Lexicographic order on a sigma type
+
+This defines the lexicographical order of two arbitrary relations on a sigma type and proves some
+lemmas about `PSigma.Lex`, which is defined in core Lean.
+
+Given a relation in the index type and a relation on each summand, the lexicographical order on the
+sigma type relates `a` and `b` if their summands are related or they are in the same summand and
+related by the summand's relation.
+
+## See also
+
+--Porting note: fix these once the files are ported.
+Related files are:
+* `data.finset.colex`: Colexicographic order on finite sets.
+* `data.list.lex`: Lexicographic order on lists.
+* `data.sigma.order`: Lexicographic order on `Σ i, α i` per say.
+* `data.psigma.order`: Lexicographic order on `Σ' i, α i`.
+* `data.prod.lex`: Lexicographic order on `α × β`. Can be thought of as the special case of
+  `sigma.lex` where all summands are the same
```

Comment:
```suggestion
Related files are:
* `Data.Finset.Colex`: Colexicographic order on finite sets.
* `Data.List.Lex`: Lexicographic order on lists.
* `Data.Sigma.Order`: Lexicographic order on `Σ i, α i` per say.
* `Data.PSigma.Order`: Lexicographic order on `Σ' i, α i`.
* `Data.Prod.Lex`: Lexicographic order on `α × β`. Can be thought of as the special case of
  `Sigma.Lex` where all summands are the same
```
I am not sure whether it is `Colex` or `CoLex`, but all other names are easy and I think we should change them now. Also that file seems to non-existing in mathlib3 (maybe moved to `combinatorics.colex`?)